### ISSUE TITLE
Modelica.Media.Air.MoistAir.isentropicEnthalpyApproximation not working

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#952]
**Reported by mtiller on 3 Jan 2013 16:21 UTC**
This is related to ticket #950, but a different issue.  If I try to use the sample model from that ticket but utilize isentropicEnthalpyApproximation instead, I get another (different) error.  The code to trigger this issue is:

```modelica
model TestIsentropicEnthalpy
  package Medium = Modelica.Media.Air.MoistAir;
  Medium.ThermodynamicState state(p=Medium.p_default,T=Medium.T_default,X=Medium.X_default);
  Medium.SpecificEnthalpy h = Medium.isentropicEnthalpyApproximation(Medium.p_default,state);
end TestIsentropicEnthalpy;
```

Dymola gives the following error at RUNTIME:

```
Log-file of program ./dymosim
(generated: Thu Jan 03 11:18:10 2013)

dymosim started
... "dsin.txt" loading (dymosim input file)
RealMatchingSizes(a,b)
The following error was detected at time: 0
Array dimensions did not match
The stack of functions is:
Modelica.Media.Air.MoistAir.isentropicEnthalpyApproximation
Modelica.Media.Air.MoistAir.isentropicEnthalpyApproximation(
101325, 
Modelica.Media.Air.MoistAir.ThermodynamicState(
p = 101325, 
T = 293.15, 
X = {0.01, 0.99}
))
Error: Failed to start model.
```

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/952

**Comment by modelica-trac-importer** [comment]
**Comment by hubertus on 3 Jan 2013 16:55 UTC**
Yes, I recognize this. This has to do with the sort of insane idea that someone wanted to keep 2 options in the interface, that in combination are confusing and error prone: both N and N-1 are allowed as vector lengths for the composition vector, governed by an option. I am the library officer, but I don't feel responsible for this design decision, which I consider flawed. I will do my best to fix this instance of the problem. It is not the first, I would be surprised this is the last one to be found.

**Comment by modelica-trac-importer** [comment]
**Comment by hubertus on 6 Mar 2013 14:27 UTC**
Fixed in 9aa3a06ecc802d84ab7a38ea59eafcd904a9851d. Closing.
